yjit/src/utils.rs

```diff
@@ -76,6 +76,9 @@ pub(crate) use offset_of;
 fn ruby_str_to_rust(v: VALUE) -> String {
     // Make sure the CRuby encoding is UTF-8 compatible
     let encoding = unsafe { rb_ENCODING_GET(v) } as u32;
+    if !(encoding == RUBY_ENCINDEX_ASCII_8BIT || encoding == RUBY_ENCINDEX_UTF_8 || encoding == RUBY_ENCINDEX_US_ASCII) {
+        return String::new();
+    }
```

Comment:
A bit confused about this bit since it seems that would return an empty string for some encodings?

Reply:
The current branch has a lot of experimental code that I wrote with Alan during pairing. It was a temporary workaround to run entire test-spec with `--yjit-dump-disasm=all` for debugging. I'll remove it when I clean up this branch again. We should be able to run it without writing a workaround every time, but it's not a scope of this PR.